lib/matplotlib/widgets.py

```diff
@@ -1934,8 +1934,15 @@ def onmove(self, event):
             self.linev.set_visible(False)
             self.lineh.set_visible(False)
             if self.needclear:
+                if self.useblit and self.background is not None:
+                    self.canvas.restore_region(self.background)
+                else:
+                    self.canvas.draw()
+                    self.canvas.restore_region(self.background)
```

Comment:
`self.background` may be `None` here, in which case this fails.  I think it is always `None` if `useblit` is `False`.